hs_discover/views.py

```diff
@@ -1,57 +1,224 @@
+import datetime
 import json
+import time
+import logging
+from collections import namedtuple
 
+from django.core.paginator import Paginator
+from django.http import JsonResponse
 from django.shortcuts import render
 from django.views.generic import TemplateView
 from haystack.query import SearchQuerySet
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+logger = logging.getLogger(__name__)
+
+DateRange = namedtuple('DateRange', ['start', 'end'])
 
 
-# TODO error handling, validation, analytics
 class SearchView(TemplateView):
 
     def get(self, request, *args, **kwargs):
-        from django.template.defaultfilters import date, time
-        q = request.GET.get('q') if request.GET.get('q') else ""
+        return render(request, 'hs_discover/index.html', {})
+
+
+class SearchAPI(APIView):
+
+    def __init__(self, **kwargs):
+        super(SearchAPI, self).__init__(**kwargs)
+        self.filterlimit = 20
+        self.perpage = 35
+
+    def get(self, request, *args, **kwargs):
+        """
+        Primary endpoint for retrieving resources via the index
+        Values should never be empty string or python None, instead return string "None" with str() call
+        "availability": list value, js will parse JSON as Array
+        "availabilityurl": single value, pass a string to REST client
+        "type": single value, pass a string to REST client
+        "author": single value, pass a string to REST client first author
+        "creator: authors,
+                The reason for the weird name is the DataOne standard. The metadata was designed to be compliant
+                with DataOne standards. These standards do not contain an author field. Instead, the creator field
+                represents authors.
+        "contributor": list value, js will parse JSON as Array
+        "owner": list value, js will parse JSON as Array
+        "subject": list value, js will parse JSON as Array
+        "coverage_type": list point, period, ...
+        """
+        start = time.time()
+
+        sqs = SearchQuerySet().all()
+
+        if request.GET.get('filterbuilder'):
```

Comment:
I think you asked me earlier whether this should be broken out to it's own endpoint, and I do think it should.  If it's too much of a bother to do now, maybe it's worth just adding a TODO here?

Reply:
The TODO is a larger concern and planning around urls.py , NGINX and the sub-urls plan for REST and API clients. We should capture items like these elsewhere, as it loses its context and meaning to use this one of many locations to track that.

Reply:
ok, log an issue or handle this however you believe is appropriate, then resolve this conversation.

Reply:
https://github.com/hydroshare/hydroshare/issues/3988